src/nxt_otel.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) F5, Inc.
+ */
+
+#ifndef _NXT_OTEL_H_INCLUDED_
+#define _NXT_OTEL_H_INCLUDED_
+
+#include <nxt_router.h>
+#include <nxt_string.h>
+
+// forward declared
+struct nxt_http_field_t;
+struct nxt_conf_validation_t;
+struct nxt_conf_value_t;
+struct nxt_http_request_t;
+
+extern void     nxt_otel_rs_send_trace(void *trace);
+extern void   * nxt_otel_rs_get_or_create_trace(u_char *trace_id);
+extern void     nxt_otel_rs_init(void (*log_callback)(u_char *log_string),
+                                 const nxt_str_t *endpoint,
+                                 const nxt_str_t *protocol,
+                                 double sample_fraction,
+                                 double batch_size);
+extern void     nxt_otel_rs_copy_traceparent(u_char *buffer, void *span);
+extern void     nxt_otel_rs_add_event_to_trace(void *trace,
+                                               nxt_str_t *key,
+                                               nxt_str_t *val);
+extern uint8_t  nxt_otel_rs_is_init(void);
+extern void     nxt_otel_rs_uninit(void);
+
+
+/* nxt_otel_status_t
+ * more efficient than a single handler state struct
+ */
+typedef enum {
+    NXT_OTEL_UNINIT_STATE = 0,
+    NXT_OTEL_INIT_STATE,
+    NXT_OTEL_HEADER_STATE,
+    NXT_OTEL_BODY_STATE,
+    NXT_OTEL_COLLECT_STATE,
+    NXT_OTEL_ERROR_STATE,
+} nxt_otel_status_t;
+
+/* nxt_otel_state_t
+ * cache of trace data needed per request and
+ * includes indicator as to current flow state
+ */
+typedef struct {
+    u_char            *trace_id;
+    u_char            *version;
+    u_char            *parent_id;
+    u_char            *trace_flags;
+    void              *trace;
+    nxt_otel_status_t status;
+    nxt_str_t         trace_state;
+} nxt_otel_state_t;
```

Comment:
Our preferred way (latterly) is to split the typedef out from the definition. E.g.

The above would become

```c
typedef enum nxt_otel_status_e   nxt_otel_status_t;
typedef struct nxt_otel_state_s  nxt_otel_state_t;
...
enum nxt_otel_status_e {
...
};

struct nxt_otel_state_s {
...
};
```

To quote Alex

> This allows using the type in declarations before it's actually
defined, and also to move the typedef to another file.